OSSMDOC-168: Part 1 of Kiali config reference. (#42168)
openshift/openshift-docs: modules/ossm-config-smcp-kiali.adoc

// Module included in the following assemblies:
//* service_mesh/v2x/ossm-reference-kiali.adoc
:_content-type: REFERENCE
[id="ossm-smcp-kiali_{context}"]
= Specifying Kiali configuration in the SMCP

You can configure Kiali under the `addons` section of the `ServiceMeshControlPlane` resource. Kiali is enabled by default. To disable Kiali, set `spec.addons.kiali.enabled` to `false`.

You can specify your Kiali configuration in either of two ways:

> * Specify the Kiali configuration in the `ServiceMeshControlPlane` resource under `spec.addons.kiali.install`. There are some limitations with this approach, as the complete list of Kiali configurations is not available in the SMCP.

Reviewer: ```suggestion
* Specify the Kiali configuration in the ServiceMeshControlPlane resource under spec.addons.kiali.install. This approach has some limitations, because the complete list of Kiali configurations is not available in the SMCP.
```

Reviewer: IBM SG: as adverb
Do not use to mean "because" or "while".
Example: Write "Because you created the file, you can delete it", not "As you created the file, you can delete
it".

|`true`

|kiali:
  install:
|Install a Kiali resource if the named Kiali resource is not present. The `install` section is ignored if `addons.kiali.enabled` is set to `false`.
|
|

|kiali:
  install:
    dashboard:
|Configuration parameters for the dashboards shipped with Kiali.
|
|

|kiali:
  install:
    dashboard:
      viewOnly:
|This parameter enables/disables view-only mode for the Kiali console. When view-only mode is enabled, users cannot use the Kiali console to make changes to the {SMProductShortname}.
|`true`/`false`
|`false`

|kiali:
  install:
    dashboard:
      enableGrafana:
|Grafana endpoint configured based on `spec.addons.grafana` configuration.
|`true`/`false`
|`true`

|kiali:
  install:
    dashboard:
      enablePrometheus:
|Prometheus endpoint configured based on `spec.addons.prometheus` configuration.
|`true`/`false`
|`true`

|kiali:
  install:
    dashboard:
      enableTracing:
|Tracing endpoint configured based on Jaeger custom resource configuration.
|`true`/`false`
|`true`

|kiali:
  install:
    service:
|Configuration parameters for the Kubernetes service associated with the Kiali installation.
|
|

|kiali:
  install:
    service:
      metadata:

> |Use to specify additional metadata to apply to resources.

Reviewer: ```suggestion
|Specifies additional metadata to apply to resources.
```
or 
```suggestion
|Additional metadata to apply to resources.
```

Reviewer: Multiple instances.

|N/A
|N/A

|kiali:
  install:
    service:
      metadata:
        annotations:

> |Use to specify additional annotations to be applied to the component's service.

Reviewer: ```suggestion
|Use to specify additional annotations to apply to the component's service.
```
Multiple instances.

|string
|N/A

|kiali:
  install:
    service:
      metadata:
        labels:
|Use to specify additional labels to be applied to the component's service.
|string
|N/A

|kiali:
  install:
    service:
      ingress:

> |Used to specify details for accessing the component’s service through an OpenShift Route.

Reviewer: ```suggestion
|Use to specify details for accessing the component’s service through an OpenShift Route.
```
to be consistent with preceding instances.

|N/A
|N/A

|kiali:
  install:
    service:
      ingress:
        metadata:
          annotations:
|Use to specify additional annotations to be applied to the component's service ingress.
|string
|N/A

|kiali:
  install:
    service:
      ingress:
        metadata:
          labels:
|Use to specify additional labels to be applied to the component's service ingress.
|string
|N/A

|kiali:
  install:
    service:
      ingress:
        enabled:
|Use to customize an OpenShift Route for the service associated with a component.
|`true`/`false`
|`true`

|kiali:
  install:
    service:
      ingress:
        contextPath:
|Use to specify the context path to the service.
|string
|N/A

|install:
  service:
    ingress:
      hosts:

> |Use to specify a single host name per OpenShift route. An empty host name implies a default host name for the Route.

Reviewer: ```suggestion
|Use to specify a single hostname per OpenShift route. An empty hostname implies a default host name for the Route.
```
Multiple instances.
IBM SG: hostname noun
Write as shown unless updating existing content that uses "host name".

// Module included in the following assemblies:
//* service_mesh/v2x/ossm-reference-kiali.adoc
:_content-type: REFERENCE
[id="ossm-smcp-kiali_{context}"]
= Specifying Kiali configuration in the SMCP

You can configure Kiali under the `addons` section of the `ServiceMeshControlPlane` resource. Kiali is enabled by default. To disable Kiali, set `spec.addons.kiali.enabled` to `false`.

You can specify your Kiali configuration in either of two ways:

* Specify the Kiali configuration in the `ServiceMeshControlPlane` resource under `spec.addons.kiali.install`. There are some limitations with this approach, as the complete list of Kiali configurations is not available in the SMCP.

> * Configure and deploy a Kiali instance and specify the name of the Kiali resource as the value for `spec.addons.kiali.name` in the `ServiceMeshControlPlane` resource. You must create the CR in the same namespace as the Service Mesh control plane, for example, `istio-system`. If a Kiali resource matching the value of `name` exists, the control plane will use the existing installation. This approach lets you fully customize your Kiali configuration in the Kiali resource.  Note that with this approach, various fields in the Kiali resource are overwritten by the {ProductShortName} Operator, specifically, the `accessible_namespaces` list, as well as the endpoints for Grafana, Prometheus, and tracing.

Reviewer 2: This is actually a cool feature that I did not know about. I assume QE tested this feature :)

Author: I don't know, we'll have to ask them.  I've played with it a couple of times and I still need to figure out how things work so that I've got the description correct.

.Example SMCP parameters for Kiali
[source,yaml]
----
apiVersion: maistra.io/v2
kind: ServiceMeshControlPlane
metadata:
  name: basic
spec:
  addons:
    kiali:
      name: kiali
      enabled: true
      install:
        dashboard:
          viewOnly: false
          enableGrafana: true
          enableTracing: true
          enablePrometheus: true
        service:
          ingress:
            contextPath: /kiali
----

.Kiali parameters in the SMCP
[options="header"]
[cols="l, a, a, a"]
|===
|Parameter |Description |Values |Default value
|spec:
  addons:
    kiali:
      name:
|Name of Kiali custom resource. If a Kiali CR matching the value of `name` exists, the {ProductShortname} Operator will use that CR for the installation. If no Kiali CR exists, the Operator will create one using this `name` and the configuration options specified in the SMCP.
|string
|`kiali`

|kiali:
  enabled:
|This parameter enables/disables Kiali. Kiali is enabled by default.
|`true`/`false`
|`true`

|kiali:
  install:

> |Install a Kiali resource if the named Kiali resource is not present.

Reviewer 2: Shouldn't you say something like, "This `install` section is ignored if `addons.kiali.enabled` is `false`" ??

|`true`/`false`
|`false`

|kiali:
  install:
    dashboard:
      enableGrafana:
|Grafana endpoint configured based on `spec.addons.grafana` configuration.
|`true`/`false`
|`true`

|kiali:
  install:
    dashboard:
      enablePrometheus:
|Prometheus endpoint configured based on `spec.addons.prometheus` configuration.
|`true`/`false`
|`true`

|kiali:
  install:
    dashboard:
      enableTracing:
|Tracing endpoint configured based on Jaeger custom resource configuration.
|`true`/`false`
|`true`

|kiali:
  install:
    service:
|Configuration parameters for the Kubernetes service associated with the Kiali installation.
|
|

|kiali:
  install:
    service:
      metadata:
|Use to specify additional metadata to apply to resources.
|N/A
|N/A

|kiali:
  install:
    service:
      metadata:
        annotations:
|Use to specify additional annotations to be applied to the component's service.
|string
|N/A

|kiali:
  install:
    service:
      metadata:
        labels:
|Use to specify additional labels to be applied to the component's service.
|string
|N/A

|kiali:
  install:
    service:
      ingress:

> |Used to specify details for accessing the component’s service through a Kubernetes Ingress or an OpenShift Route.

Reviewer 2: Because these docs are within the context of OpenShift, remove the "a Kubernetes Ingress or" phrase. When installing on OpenShift, it will install a Route, never a Kubernetes Ingress.
```suggestion
|Used to specify details for accessing the component’s service through an OpenShift Route.
```
